app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -1338,6 +1338,14 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = setOf()
     )
 
+    /**
+     * Indicates if sync tabs settings is enabled.
+     */
+    var syncOpenTabsEnabled by booleanPreference(
```

Comment:
Shouldn't we be getting this from the Account Manager?

Reply:
My original thinking is that this shared preference exists already. Since we update the check box preferences in account settings.  But I'm seeing other issues when turning on and off sync open tabs in account settings.

Reply:
Figured out the issue but it's unreleated to this.  I'll use the account manager as suggested.  Thanks

Reply:
After some investigation, I've moved this check to much earlier so we short circuit if sync tabs is not on.

Reply:
What's nice about this change is that it happens in the coroutine flow, so when we check the SyncEnginesStorage, that change happens off the main thread and avoids any perf losses for the homescreen every time the screen is presented.

Reply:
We can also makes it easier to use his [other Synced Tabs change](https://github.com/mozilla-mobile/android-components/pull/12628) when it lands in AC.